Tests/Packet++Test/Tests/TcpTests.cpp

```diff
@@ -285,4 +286,37 @@ PTF_TEST_CASE(TcpPacketCreation2)
 	pcpp::TcpOption tcpSnackOption = tcpLayer.addTcpOption(pcpp::TcpOptionBuilder(pcpp::TCPOPT_SNACK, NULL, PCPP_TCPOLEN_SNACK));
 	PTF_ASSERT_TRUE(tcpSnackOption.isNotNull());
 	PTF_ASSERT_TRUE(tcpSnackOption.setValue(htobe32(1000)));
-} // TcpPacketCreation2
\ No newline at end of file
+} // TcpPacketCreation2
+
+
+PTF_TEST_CASE(TcpHash5Tuple)
+{
+	pcpp::MacAddress srcMac("30:46:9a:23:fb:fa");
+	pcpp::MacAddress dstMac("08:00:27:19:1c:78");
+	pcpp::EthLayer ethLayer(srcMac, dstMac, PCPP_ETHERTYPE_IP);
+	pcpp::IPv4Address dstIP("10.0.0.6");
+	pcpp::IPv4Address srcIP("212.199.202.9");
+	pcpp::IPv4Layer ipLayer(srcIP, dstIP);
+	ipLayer.getIPv4Header()->ipId = htobe16(20300);
+	ipLayer.getIPv4Header()->fragmentOffset = htobe16(0x4000);
+	ipLayer.getIPv4Header()->timeToLive = 59;
+	pcpp::TcpLayer tcpLayer((uint16_t)80, (uint16_t)44160);
+	tcpLayer.getTcpHeader()->sequenceNumber = htobe32(0xb829cb98);
+	tcpLayer.getTcpHeader()->ackNumber = htobe32(0xe9771586);
+	tcpLayer.getTcpHeader()->ackFlag = 1;
+	tcpLayer.getTcpHeader()->pshFlag = 1;
+	tcpLayer.getTcpHeader()->windowSize = htobe16(20178);
+	tcpLayer.getTcpHeader()->portSrc = htobe16(60388);
+	tcpLayer.getTcpHeader()->portDst = htobe16(80);
+
+	pcpp::Packet tcpPacket(1);
+	tcpPacket.addLayer(&ethLayer);
+	tcpPacket.addLayer(&ipLayer);
+	tcpPacket.addLayer(&tcpLayer);
+
+	tcpPacket.computeCalculateFields();
+
+        PTF_ASSERT_EQUAL(pcpp::hash5Tuple(&tcpPacket), 1576639238, u32);
+        PTF_ASSERT_EQUAL(pcpp::hash5Tuple(&tcpPacket, true), 2243556734, u32);
```

Comment:
Another test you can add is: build another packet with opposite src and dst and check that `hash5Tuple` with `directionunique = false` gives the same hash value while `directionunique = true` gives a different value